src/shell/interpreter.zig

```diff
@@ -10101,6 +10114,734 @@ pub const Interpreter = struct {
+                });
+                if (err_) |err| {
+                    this.state.exec.err = err;
+                    const error_string = this.bltn.taskErrorToString(.cp, err);
+                    output_task.start(error_string);
+                    return;
+                }
+                output_task.start(null);
+            }
+
+            pub const ShellCpOutputTask = OutputTask(Cp, .{
+                .writeErr = ShellCpOutputTaskVTable.writeErr,
+                .onWriteErr = ShellCpOutputTaskVTable.onWriteErr,
+                .writeOut = ShellCpOutputTaskVTable.writeOut,
+                .onWriteOut = ShellCpOutputTaskVTable.onWriteOut,
+                .onDone = ShellCpOutputTaskVTable.onDone,
+            });
+
+            const ShellCpOutputTaskVTable = struct {
+                pub fn writeErr(this: *Cp, childptr: anytype, errbuf: []const u8) CoroutineResult {
+                    if (this.bltn.stderr.needsIO()) {
+                        this.state.exec.output_waiting += 1;
+                        this.bltn.stderr.enqueue(childptr, errbuf);
+                        return .yield;
+                    }
+                    _ = this.bltn.writeNoIO(.stderr, errbuf);
+                    return .cont;
+                }
+
+                pub fn onWriteErr(this: *Cp) void {
+                    this.state.exec.output_done += 1;
+                }
+
+                pub fn writeOut(this: *Cp, childptr: anytype, output: *OutputSrc) CoroutineResult {
+                    if (this.bltn.stdout.needsIO()) {
+                        this.state.exec.output_waiting += 1;
+                        this.bltn.stdout.enqueue(childptr, output.slice());
+                        return .yield;
+                    }
+                    _ = this.bltn.writeNoIO(.stdout, output.slice());
+                    return .cont;
+                }
+
+                pub fn onWriteOut(this: *Cp) void {
+                    this.state.exec.output_done += 1;
+                }
+
+                pub fn onDone(this: *Cp) void {
+                    this.next();
+                }
+            };
+
+            pub const ShellCpTask = struct {
+                cp: *Cp,
+
+                opts: Opts,
+                operands: usize = 0,
+                src: [:0]const u8,
+                tgt: [:0]const u8,
+                src_absolute: ?[:0]const u8 = null,
+                tgt_absolute: ?[:0]const u8 = null,
+                cwd_path: [:0]const u8,
+                verbose_output_lock: std.Thread.Mutex = .{},
+                verbose_output: ArrayList(u8) = ArrayList(u8).init(bun.default_allocator),
+
+                task: JSC.WorkPoolTask = .{ .callback = &runFromThreadPool },
+                event_loop: JSC.EventLoopHandle,
+                concurrent_task: JSC.EventLoopTask,
+                err: ?bun.shell.ShellErr = null,
+
+                const print = bun.Output.scoped(.ShellCpTask, false);
```

Comment:
we don't use `print` anywhere else in the codebase for these

can this be called `debug` or `log`?